### DISCUSSION TITLE
Main branch has been renamed to.. main

### DISCUSSION THREAD

**Discussion by BJReplay** [BJReplay/ha-solcast-solar#117] [Announcements]
As the title says, I have bitten the bullet, and renamed the main branch to main

Assuming that you have a git command line client that you are familiar with, this is what you need to do to get back in sync:

```
git branch -m v4.1 main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```

**Comment by autoSteve** [comment]
Resume normal programming on desktop… Except web has had a stroke:

 

 

 

 

What the actual FOSS, GitHub?

 

 

From: BJReplay ***@***.***>
Reply to: BJReplay/ha-solcast-solar ***@***.***>
Date: Friday 9 August 2024 at 19:11
To: BJReplay/ha-solcast-solar ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [BJReplay/ha-solcast-solar] Main branch has been renamed to.. main (Discussion #117)

 

As the title says, I have bitten the bullet, and renamed the main branch to main

Assuming that you have a git command line client that you are familiar with, this is what you need to do to get back in sync:
git branch -m v4.1 main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a

**Reply by BJReplay** [reply]
Tagging

@isorin 
@iainfogg
@gcoan

As contributors, this change may impact you.

**Reply by gcoan** [reply]
Not got to grips with the command line, tried to manually do a git merge on another repository and just got nowhere.

But back to solcast, I assume we can just re-fork the bjreplay solcast repository?

**Reply by autoSteve** [reply]
You shouldn't need to re-fork.

Rename your default branch in UI? 🤷‍♂️

<img width="1378" alt="image" src="https://github.com/user-attachments/assets/96859839-f4bb-4727-95d1-1ba6739e7d5c">

**Reply by gcoan** [reply]
'twas as easy as that, github even noticed that the root branch had been renamed and prompted me to rename my fork to match

![image](https://github.com/user-attachments/assets/85584be7-f451-47e7-a57a-af87d9406e4b)

Oh so much easier than command line nerding around

**Reply by autoSteve** [reply]
Is it _wrong_ to want a GUI? To fix mine it was a small matter of a hard reset to the remote head's most recent commit after rename. (Said no one sane ever.) My fork stubbornly thought it was one commit ahead, and this was the only way to fix it. No amount of GUI-ing would solve it...

Glad you're up-to-date.

```
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
git fetch upstream
git reset --hard upstream/main
git push --force
```

**Comment by BJReplay** [comment]
Closing - this happened long enough ago that people who need to know already know.
